ui/components-react/src/components-react/filter-builder/FilterBuilderRuleValue.tsx

```diff
@@ -0,0 +1,36 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+import * as React from "react";
+import { PropertyDescription, PropertyRecord, PropertyValue, PropertyValueFormat } from "@itwin/appui-abstract";
+import { EditorContainer, PropertyUpdatedArgs } from "../editors/EditorContainer";
+
+/** @alpha */
+export interface FilterBuilderRuleValueProps {
+  value?: PropertyValue;
+  property: PropertyDescription;
+  onChange: (value: PropertyValue) => void;
+}
+
+/** @alpha */
+export function FilterBuilderRuleValue(props: FilterBuilderRuleValueProps) {
+  const {value, property, onChange} = props;
+
+  const propertyRecord = React.useMemo(() => {
+    return new PropertyRecord(value ?? {valueFormat: PropertyValueFormat.Primitive}, property);
+  }, [value, property]);
+
+  const onValueChange = React.useCallback(({newValue}: PropertyUpdatedArgs) => {
+    onChange(newValue);
```

Comment:
I am not sure if it's worth. It looks like it will be not enough to check references because `EditorContainer` invokes `onCommit` every time with new value object. So that check should look something like this:

`
if (newValue.valueFormat === PropertyValueFormat.Primitive && value?.valueFormat === PropertyValueFormat.Primitive
      && newValue.value === value.value && newValue.displayValue === value.displayValue)
      return;
`

Reply:
agree - updating the value on our component side doesn't look that expensive to start worrying about this